src/api/context.ts

```diff
@@ -0,0 +1,65 @@
+/*********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import Command from '@oclif/command'
+import Listr = require('listr')
+import * as os from 'os'
+import * as path from 'path'
+
+import { CHE_OPERATOR_CR_PATCH_YAML_KEY, CHE_OPERATOR_CR_YAML_KEY, LOG_DIRECTORY_KEY } from '../common-flags'
+import { readCRFile } from '../util'
+
+import { KubeHelper } from './kube'
+
+/**
+ * chectl context.
+ */
+export namespace ChectlContext {
+  export const IS_OPENSHIFT = 'isOpenShift'
+  export const IS_OPENSHIFT4 = 'isOpenShift4'
+  export const START_TIME = 'startTime'
+  export const END_TIME = 'endTime'
+  export const CONFIG_DIR = 'configDir'
+  export const ERROR_LOG = 'errorLog'
+  export const COMMAND_ID = 'commandId'
+  export const CUSTOM_CR = 'customCR'
+  export const CR_PATCH = 'crPatch'
```

Comment:
I do not think that these properties are part of chectl context. They are present for some group of installers only.

Reply:
Yes, I don't see any problems with that.

Reply:
Then, maybe, we have to have a single context?